### ISSUE TITLE
[20.0] ERROR: no matching manifest for linux/386 in the manifest list entries

### ISSUE THREAD

**Issue by bendeg** [georchestra/docker#59]
Hi there,

I could use compose up, it pulled everything and stopped at the end with this error.

Is this a problem of 32 bits architecture (and should be 64) ?
Or do I missed something ?

**Comment by bendeg** [comment]
docker info  -f '{{.OSType}}/{{.Architecture}}' gives me : 

linux/i686

**Comment by bendeg** [comment]
docker version gives me : 

Client:
 Version:           19.03.7
 API version:       1.40
 Go version:        go1.14.3
 Git commit:        7141c19
 Built:             Sun, 17 May 2020 17:07:15 +1000
 OS/Arch:           linux/386
 Experimental:      false

Server:
 Engine:
  Version:          19.03.7
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.14.3
  Git commit:       7141c19
  Built:            Sun May 17 07:07:15 2020
  OS/Arch:          linux/386
  Experimental:     false
 containerd:
  Version:          19.03.7
  GitCommit:        7c1e88399ec0b0b077121d9d5ad97e647b11c870
 runc:
  Version:          1.0.0~rc10+dfsg2
  GitCommit:        1.0.0~rc10+dfsg2-1
 docker-init:
  Version:          0.18.0
  GitCommit:

**Comment by fvanderbiest** [comment]
> I could use compose up, it pulled everything and stopped at the end with this error.

Which error is it ?

**Comment by RemiDesgrange** [comment]
geOrchestra can run on 32bit architecture (theoretically, you know, Java's mantra "Write Once, Run Everywhere". But we are building docker images on 64bits machine, which pull 64bits JVM. So you cannot run these images on 32 bits architecture. 
When I look at the tags of [jetty image on docker hub](https://hub.docker.com/_/jetty?tab=tags&page=1&name=9-jre8) I can see that there are no images for the tag 9-jre8 with x86_32. If you manage to get it working, it could be nice to have your input.

**Comment by bendeg** [comment]
I'm new to docker, but I think there's one available on docker hub :  9-jre8-alpine tagged for linux/386 (the 4th one in the list). Maybe I'm wrong but it's the right architecture to run on a 32 bits machine, right ?
If yes, do you mean I have to create a new docker-compose.yml file whit that version ? (meaning only changing that container)
[EDIT : I think I'm far from it, all images have to be build for 32 bits first and pushed on Docker hub...]
Anyway, I tried georchestra 20.0 on a 64bits machine and it works fine.

**Comment by bendeg** [comment]
I tried a docker image build (analytics) :

1) In analytics Dockerfile : I changed **FROM jetty:9-jre8** into **FROM jetty:9-jre8-alpine**
It works and pulls the image.

2) build stops saying he doesn't find **/etc/georchestra** (step 9 of build, at line **RUN chown jetty:jetty /etc/georchestra** I guess...)

3) **/etc/georchestra** was indeed not in my system, so a gitcloned it in /etc/georchestra (with root account)

4) build still stops saying he doesn't find /etc/georchestra (**chown: /etc/georchestra: No such file or directory**)

What did I wrong ?

**Comment by fvanderbiest** [comment]
1. Why relying on alpine ? You're on your own, doing so ...

2. How do you build ?

**Comment by bendeg** [comment]
Sorry, I should have look at your build instructions first...(make docker-build)

I tried to build all needed images (found in docker-compose.yml) one by one...

Trying make docker-build now...Man, it's scary :-), hope it'll work...

**Comment by bendeg** [comment]
**Images (19.04) build successfuly** : 

camptocamp/courier-imap:latest
camptocamp/smtp-sink:latest
 camptocamp/sqwebmail:latest
georchestra/geonetwork:19.04
georchestra/ssh_data:latest

For georchestra/geonetwork:19.04, I had to change **FROM jetty:9-jre8** to **FROM jetty:9-jre8-alpine** in Dockerfile otherwise it fails => I changed every Dockerfile that contain FROM jetty:9-jre8


**georchestra/geoserver:19.04 build failed** : 

**Step 6/12** : RUN echo "deb http://httpredir.debian.org/debian stretch main contrib" > /etc/apt/sources.list     && echo "deb http://security.debian.org/ stretch/updates main contrib" >> /etc/apt/sources.list     && echo "ttf-mscorefonts-installer msttcorefonts/accepted-mscorefonts-eula select true" | debconf-set-selections     && apt-get update     && apt-get install -y ttf-mscorefonts-installer     && apt-get clean     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*

 ---> Running in f364196da460
**/bin/sh: can't create /etc/apt/sources.list: nonexistent directory**

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  42.576 s
[INFO] Finished at: 2020-07-02T13:50:47+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.3.8:build (default-cli) on project geoserver-webapp: Exception caught: The command '/bin/sh -c echo "deb http://httpredir.debian.org/debian stretch main contrib" > /etc/apt/sources.list     && echo "deb http://security.debian.org/ stretch/updates main contrib" >> /etc/apt/sources.list     && echo "ttf-mscorefonts-installer msttcorefonts/accepted-mscorefonts-eula select true" | debconf-set-selections     && apt-get update     && apt-get install -y ttf-mscorefonts-installer     && apt-get clean     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 1 -> [Help 1]

**Comment by fvanderbiest** [comment]
> /bin/sh: can't create /etc/apt/sources.list: nonexistent directory

See https://wiki.alpinelinux.org/wiki/Alpine_Linux_package_management

**Comment by bendeg** [comment]
It's working now, with jetty **9.4.9-jre8** image (thanks !)

I think all images related to docker-build are present : 

$docker images 
REPOSITORY                   TAG                 IMAGE ID            CREATED             SIZE
georchestra/header           19.04               9ac5568ef660        18 seconds ago      455MB
georchestra/analytics        19.04               37866b2fc1a1        2 minutes ago       469MB
georchestra/security-proxy   19.04               008cd6f7dd22        11 minutes ago      464MB
georchestra/atlas            19.04               5c19f97633fc        14 minutes ago      537MB
georchestra/mapfishapp       19.04               74aacafee72e        29 minutes ago      516MB
georchestra/console          19.04               d5fd0ee0fb14        49 minutes ago      484MB
georchestra/geowebcache      19.04               479e4317e4ed        About an hour ago   448MB
georchestra/extractorapp     19.04               acd9bfae4daf        About an hour ago   685MB
georchestra/cas              19.04               4fdabab09709        2 hours ago         473MB
georchestra/geoserver        19.04               4b041ac793be        2 hours ago         566MB
georchestra/geonetwork       19.04               400390069984        2 hours ago         687MB
georchestra/ldap             19.04               f5385d4417fe        2 hours ago         155MB
georchestra/database         19.04               82443231fc9b        2 hours ago         427MB
georchestra/ssh_data         latest              019e5ab58ec6        2 hours ago         193MB
camptocamp/sqwebmail         latest              e8fae08847c7        2 hours ago         269MB
camptocamp/courier-imap      latest              c98cc429a549        2 hours ago         298MB
camptocamp/smtp-sink         latest              7a6bd069a9ec        2 hours ago         316MB
debian                       stretch             f131b82bb4ad        3 weeks ago         101MB
jetty                        9.4.9-jre8          6b8e9868c2ce        2 years ago         448MB

**Comment by fvanderbiest** [comment]
:+1:

**Comment by bendeg** [comment]
Build was indeed successful but : 

docker-compose needs TRAEFIK image which, again, is not available for 386 architecture.

So I used bibliosansfrontieres/traefik-i386 and docker-compse up starts.
I can access http://georchestra.mydomain.org.
Login works.
Viewer works.
Administration modules work.

But Geoserver and Catalog don't work (it's waiting forever...).

Logs : 

2020-07-02 15:23:37 security [ERROR] Unknown host: georchestra.mydomain.org
georchestra.mydomain.org_1  | time="2020-07-02T14:59:23Z" level=warning msg="Error checking new version: Get https://update.traefik.io/repos/containous/traefik/releases: x509: certificate signed by unknown authority"
proxy_1                     | 2020-07-02 15:00:27 security [ERROR] Exception occured when trying to connect to the remote host: 
proxy_1                     | java.util.concurrent.TimeoutException
etc...


geonetwork_1                | Invalid maximum heap size: -Xmx6G
geonetwork_1                | The specified size exceeds the maximum representable size.
geonetwork_1                | Error: Could not create the Java Virtual Machine.
geonetwork_1                | Error: A fatal exception has occurred. Program will exit.

I tried other 32 versions of traefik, it's even worse...
